src/live_restore/live_restore_fs.c

```diff
@@ -886,17 +885,25 @@ __wti_live_restore_fs_fill_holes(WT_FILE_HANDLE *fh, WT_SESSION *wt_session)
     WT_TIMER timer;
     uint64_t msg_count = 0;
     bool finished = false;
+    WT_ITEM buf;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+    WT_CLEAR(buf);
+
+    WT_RET(
+      __wt_buf_grow(session, &buf, ((WTI_LIVE_RESTORE_FS *)S2C(session)->file_system)->read_size));
 
     __wt_timer_start((WT_SESSION_IMPL *)wt_session, &timer);
     while (!finished) {
-        WTI_WITH_LIVE_RESTORE_EXTENT_LIST_WRITE_LOCK((WT_SESSION_IMPL *)wt_session,
-          (WTI_LIVE_RESTORE_FILE_HANDLE *)fh,
-          ret = __live_restore_fill_hole(fh, wt_session, &timer, &msg_count, &finished));
-        WT_RET(ret);
-        WT_RET(WT_SESSION_CHECK_PANIC(wt_session));
+        WTI_WITH_LIVE_RESTORE_EXTENT_LIST_WRITE_LOCK(session, (WTI_LIVE_RESTORE_FILE_HANDLE *)fh,
+          ret = __live_restore_fill_hole(fh, wt_session, &buf, &timer, &msg_count, &finished));
+        WT_ERR(ret);
+        WT_ERR(WT_SESSION_CHECK_PANIC(wt_session));
```

Comment:
These two lines had me pause while I figured out the 2nd one. Please add a small comment.

Reply:
[79ace73](https://github.com/wiredtiger/wiredtiger/pull/11430/commits/79ace73cb962fb5a6c60e99789a50aabc4ccf712)